Fixup Flags

rhatdan opened the pull request:
Mark hidden all references to signature-policy
Default all uses of --authfile
Add --authfile support to podman run and podman create.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

TomSweeneyRedHat commented on cmd/podman/logout.go:
```diff
@@ -24,20 +24,23 @@ var (
 		},
-		Example: `podman logout docker.io
-  podman logout --authfile authdir/myauths.json docker.io
+		Example: `podman logout quay.io
```
:+1:

TomSweeneyRedHat commented:
I'm curious, why are we hiding `--signature-policy`?  I know it's not used often, but in weird corner cases might be useful.  I'm just a little concerned that we're starting to get a handful of options that might be useful on rare occasions and we won't have a secret decoder ring anywhere for them.

Other than that, the code LGTM.

rhatdan commented:
Multiple reasons:

1.  It was not fully used in all situations
2.  It is security related, I would like to eventually move to a discussion about whether or not users can override SYS Admins decisions.    If admin says only run containers signed by this signature, we should make it more difficult to override (--registries-conf) should be similar situation.  I don't want signature-policy allowed in general right now until we make some decisions on it.  
3 Signature-policy was not fully implemented/documented for all commands that could/would use it. 
4.I don't think we should carry technical dept for an options that was added mainly for CI/CD testing.

baude commented on cmd/podman/common.go:
```diff
@@ -553,7 +558,18 @@ func getAuthFile(authfile string) string {
 		return authfile
 	}
-	return os.Getenv("REGISTRY_AUTH_FILE")
+	if remote {
```
do i read this correctly that you re-enable authfile for the remote client but basically no-op it?

TomSweeneyRedHat replied:
Just on the fence if we should do a logrus.debug or maybe info here to let them know we're in remote, auth is being ignored.

TomSweeneyRedHat replied:
Ditto certs.

baude commented on cmd/podman/pull.go:
```diff
@@ -50,14 +54,10 @@ func init() {
-		flags.BoolVar(&pullCommand.TlsVerify, "tls-verify", true, "Require HTTPS and verify certificates when contacting registries")
-	}
-
+	flags.BoolVar(&pullCommand.TlsVerify, "tls-verify", true, "Require HTTPS and verify certificates when contacting registries")
```
tls-verify was supposed to be "off" for the remote client.  have you changed your mind on it? if so, can you update the google doc that we used to review this so @jwhonce and i can be consistent?

rhatdan replied:
I don't remember why this was decided?  I am fine with turning it off, but I still think we need to be consistent between rootless and remote, (Except for when it makes little sense.)  I can turn this back off for remote, to get this PR in.  And then we can discuss at these `security` options, for non privileged users.

baude commented on cmd/podman/push.go:
```diff
@@ -35,31 +35,33 @@ var (
+	flags.StringVar(&pushCommand.Authfile, "authfile", getAuthFile(""), "Path of the authentication file. Use REGISTRY_AUTH_FILE environment variable to override")
+	flags.BoolVar(&pushCommand.Compress, "compress", false, "Compress tarball image layers when pushing to a directory using the 'dir' transport. (default is same compression type as source)")
+	flags.StringVar(&pushCommand.SignaturePolicy, "signature-policy", "", "`Pathname` of signature policy file (not usually used)")
+	flags.BoolVar(&pushCommand.TlsVerify, "tls-verify", true, "Require HTTPS and verify certificates when contacting registries")
```
again, agreement was this was a no-op for remote as well as the others.

baude commented:
Need to hit varlink stuff then too:
```
io.podman.varlink:417:    signaturePolicyPath: string,
io.podman.varlink:475:    signaturePolicyPath: string,
io.podman.varlink:766:method PushImage(name: string, tag: string, tlsverify: ?bool, signaturePolicy: string, creds: string, certDir: string, compress: bool, format: string, removeSignatures: bool, signBy: string) -> (reply: MoreResponse)
io.podman.varlink:828:method PullImage(name: string, certDir: string, creds: string, signaturePolicy: string, tlsVerify: ?bool) -> (reply: MoreResponse)
```

baude commented:
@rhatdan I'm generally in favor of this flag clean up.  For signature, would you consider removing it completely from our CLI and only support it in libpod.conf (undocumented maybe?) ?  I'd like to propose this because:

* i dont think it is used. for those that might be using it, we provide a conf way to override it.
* the above would allow it to be "global" in the sense it isn't just command to command but could be applied to the runtime.
* it isnt needed for parity with other runtimes
* now is the time to do it; just looking at the PR makes me think about the technical debt incurred by keeping it around as a "silent" option.

TomSweeneyRedHat commented:
@rhatdan thanks for the explanations.

rhatdan commented:
Baude lets discuss signature-policy at standup,  the main use case for this is CI/CD System and I don't see removing this and moving it to libpod.conf, as an easy thing to do.

rhatdan commented:
@baude Went back to not enabling these in remote commands.

Also removed cert-dir from remote commands.  Updated man pages to reflect what is not available in remote commands.

@TomSweeneyRedHat @giuseppe @baude @mheon PTAL

TomSweeneyRedHat commented:
Tests aren't hip @rhatdan, IDK if it's related to the changes or not.

mheon commented:
These aren't the tests that were blowing up on master, so I'm leaning towards not flakes

rhatdan commented:
Tests are passing
@mheon @baude @giuseppe @vrothberg @TomSweeneyRedHat PTAL

TomSweeneyRedHat commented:
LGTM, happy green test buttons.
One question for future consideration.

mheon commented:
LGTM, but I won't merge until @jwhonce and @baude can take another look and say they're satisfied - a lot of changes deep in Varlink code here

baude commented:
/lgtm
